.github/workflows/on_merge_to_main.yml

```diff
@@ -0,0 +1,181 @@
+name: Merge to main
+
+on:
+  push:
+    paths:
+      - "packages/**"
+    branches:
+      - main
+
+jobs:
+  prerelease:
+    name: Prerelease test and build
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, 'chore(release): version packages')"
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js 12.x
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12.x
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn config get cacheFolder)"
+
+      - name: Load Yarn cache
+        uses: actions/cache@v2
+        id: yarn_cache_id
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn
+
+      - name: Node modules cache
+        uses: actions/cache@v2
+        id: node_modules_cache_id
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Install Dependencies
+        if: steps.yarn_cache_id.outputs.cache-hit != 'true' || steps.node_modules_cache_id.outputs.cache-hit != 'true'
+        run: yarn install --immutable
```

Comment:
Is it possible to move this "setup" into a `job` that gets run within the other jobs, to avoid the duplication and to make these files easier to reason about?
https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#creating-a-reusable-workflow

Reply:
OMG I was looking for just this!

Reply:
OK, so this is far less useful than it seems. 

You can't share parts of a workflow. Only an entire job. Meaning you can't extend a shared workflow by adding steps to it. You can only use a shared workflow as a top level job, and that job cannot have steps. And you can't call another shared workflow from a shared workflow. 

You might be able to do that with a custom action maybe? But that seems like more work than the scope of this fix